dvc/repo/get_url.py

```diff
@@ -16,6 +16,5 @@ def get_url(url, out=None):
 
     (dep,) = dependency.loads_from(None, [url])
     (out,) = output.loads_from(None, [out], use_cache=False)
-    dep.save()
```

Comment:
It seems to me that upon `get-url` there is no point in saving the dependency. We do not have any operations related to objects that are obtained using this method. Saving dependency makes sense in case of `import`, when later we reuse the information, for example to update the asset.

Reply:
Also fixes https://github.com/iterative/dvc/issues/4251

Reply:
This makes us handle missing dependencies in a unified way though. Without it the user will get something random from the internals.

Reply:
@efiop, you mean, for example what happens upon downloading non-existient asset? Yeah that's true, but isn't `get-url` convenience method so that user does not have to work with other tools like, `wget`, `curl`? Treating it as normal dependency can limit its functionality, like in #4131. If we want to unify the handling of errors, maybe we should check if dependency exist and throw from inside the `get_url` method?

Reply:
@pared That's would a yet another place for duplicating the logic. Currently, we have unified save() and we have non-unified tree exceptions, that might vary from FileNotFound for local to something completely odd in gdrive + some unexpected behaviours. In any case, this is clearly out of scope for this weak-etag PR.

Reply:
@efiop got it, though I don't think it makes sense to keep `dep.save()` just to unify its behavior on error, because main point of `save` is to save information about the dependency. Maybe we should split functionality, so that we could `check` the dependency without actually saving it?

Reply:
@efiop I pushed another way how we can solve it, thought in this version, #4251 is no longer fixed cc @skshetry 
I think that in this case we need to decide whether we prefer to have the error handling unified, or we treated `get-url` as convenience method for the user, and can live with some ambiguous errors. Considering there is no conceptual link between `get-url` results and other commands in the project, I would lean towards the latter.

Reply:
@pared I agree, we need to unify tree exceptions and then we might not need to do anything else at all, as everything will just raise FileNotFoundError .

Reply:
For the `get` and `get-url`, is there even a need for checks? Should not `.download()` take responsibility (`download()` can depend on `tree` though)?

Reply:
Yes, potentially unified tree exceptions is all we need, but I'm not 100% sure in this particular case. The fact right now is that `download()` will raise random exceptions, which are confusing. So using `save()` for now just provides a familiar import-like and `-d` like errors, which is good enough for now.

Reply:
@pared I wonder if we can simply allow weak etags for import-url and `run -d` as well. We clearly can't use weak etags for caching, but it is good enough for dependencies. Or am I missing something?

Reply:
@pared So if we can use weak etags, could we just remove that check in this PR? The `check()` changes work, but they won't work for `import-url`, so we will get different errors, which is not very nice.

Reply:
Rolled back to allowing weak etag, #4251 is no longer fixed by this issue. cc @skshetry